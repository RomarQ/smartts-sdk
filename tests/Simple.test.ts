import SmartML from '@smartpy/smartml';
import {
    Contract,
    DefineLocal,
    EntryPoint,
    Equal,
    Flag,
    GetLocal,
    List,
    Mutez,
    Nat,
    String,
    TNat,
    Verify,
} from '../src/core';

describe('Compile Contracts', () => {
    it('Simple 1', () => {
        const contract = new Contract({
            initialStorage: Nat(1),
            entries: [
                new EntryPoint('ep1').body((arg) => [
                    DefineLocal('A', Nat(1)),
                    Verify(Equal(GetLocal('A'), arg), String('Error Message')),
                ]),
            ],
        }).toString();

        expect(contract).toMatchSnapshot();

        const michelson = SmartML.compileContract(contract);

        expect(michelson).toMatchSnapshot();
    });

    it('Simple 2', () => {
        const contract = new Contract({
            initialStorage: List([Nat(1)], TNat),
            entries: [
                new EntryPoint('ep1')
                    .config({ lazy: false })
                    .body((arg) => [
                        DefineLocal('A', Nat(1)),
                        Verify(Equal(GetLocal('A'), arg), String('Error Message')),
                    ]),
            ],
        })
            .config({
                initialBalance: Mutez(100),
                flags: [new Flag('erase-comments')],
            })
            .toString();

        expect(contract).toMatchSnapshot();

        const michelson = SmartML.compileContract(contract);

        expect(michelson).toMatchSnapshot();
    });
});
