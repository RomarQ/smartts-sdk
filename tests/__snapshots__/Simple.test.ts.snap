// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compile Contracts Simple 1 1`] = `
"
        (
            template_id (static_id 0 (\\"Simple.test.ts\\" 19))
            storage (literal (nat 1) (\\"Simple.test.ts\\" 20))
            storage_type (\\"nat\\")
            messages ((ep1 True False False True (\\"Simple.test.ts\\" 22) ((defineLocal \\"A\\" (literal (nat 1) (\\"Simple.test.ts\\" 22)) (\\"Simple.test.ts\\" 22)) (verify (eq (getLocal \\"A\\" (\\"Simple.test.ts\\" 22)) (params (\\"Simple.test.ts\\" 22)) (\\"Simple.test.ts\\" 22)) (literal (string \\"Error Message\\") (\\"Simple.test.ts\\" 22)) (\\"Simple.test.ts\\" 22)))))
            flags ()
            privates ()
            views ()
            entry_points_layout ()
            initial_metadata ()
            balance (literal (mutez 0) (\\"Simple.test.ts\\" 19))
        )
        "
`;

exports[`Compile Contracts Simple 1 2`] = `
Array [
  Object {
    "args": Array [
      Object {
        "prim": "nat",
      },
    ],
    "prim": "storage",
  },
  Object {
    "args": Array [
      Object {
        "annots": Array [
          "%ep1",
        ],
        "prim": "nat",
      },
    ],
    "prim": "parameter",
  },
  Object {
    "args": Array [
      Array [
        Object {
          "prim": "UNPAIR",
        },
        Object {
          "args": Array [
            Object {
              "prim": "nat",
            },
            Object {
              "int": "1",
            },
          ],
          "prim": "PUSH",
        },
        Object {
          "prim": "COMPARE",
        },
        Object {
          "prim": "EQ",
        },
        Object {
          "args": Array [
            Array [],
            Array [
              Object {
                "args": Array [
                  Object {
                    "prim": "string",
                  },
                  Object {
                    "string": "Error Message",
                  },
                ],
                "prim": "PUSH",
              },
              Object {
                "prim": "FAILWITH",
              },
            ],
          ],
          "prim": "IF",
        },
        Object {
          "args": Array [
            Object {
              "prim": "operation",
            },
          ],
          "prim": "NIL",
        },
        Object {
          "prim": "PAIR",
        },
      ],
    ],
    "prim": "code",
  },
]
`;

exports[`Compile Contracts Simple 2 1`] = `
"
        (
            template_id (static_id 0 (\\"Simple.test.ts\\" 37))
            storage (list (\\"Simple.test.ts\\" 38)  (literal (nat 1) (\\"Simple.test.ts\\" 38)))
            storage_type ((list \\"nat\\"))
            messages ((ep1 True False False True (\\"Simple.test.ts\\" 40) ((defineLocal \\"A\\" (literal (nat 1) (\\"Simple.test.ts\\" 42)) (\\"Simple.test.ts\\" 42)) (verify (eq (getLocal \\"A\\" (\\"Simple.test.ts\\" 42)) (params (\\"Simple.test.ts\\" 42)) (\\"Simple.test.ts\\" 42)) (literal (string \\"Error Message\\") (\\"Simple.test.ts\\" 42)) (\\"Simple.test.ts\\" 42)))))
            flags ((erase-comments))
            privates ()
            views ()
            entry_points_layout ()
            initial_metadata ()
            balance (literal (mutez 100) (\\"Simple.test.ts\\" 49))
        )
        "
`;

exports[`Compile Contracts Simple 2 2`] = `
Array [
  Object {
    "args": Array [
      Object {
        "args": Array [
          Object {
            "prim": "nat",
          },
        ],
        "prim": "list",
      },
    ],
    "prim": "storage",
  },
  Object {
    "args": Array [
      Object {
        "annots": Array [
          "%ep1",
        ],
        "prim": "nat",
      },
    ],
    "prim": "parameter",
  },
  Object {
    "args": Array [
      Array [
        Object {
          "prim": "UNPAIR",
        },
        Object {
          "args": Array [
            Object {
              "prim": "nat",
            },
            Object {
              "int": "1",
            },
          ],
          "prim": "PUSH",
        },
        Object {
          "prim": "COMPARE",
        },
        Object {
          "prim": "EQ",
        },
        Object {
          "args": Array [
            Array [],
            Array [
              Object {
                "args": Array [
                  Object {
                    "prim": "string",
                  },
                  Object {
                    "string": "Error Message",
                  },
                ],
                "prim": "PUSH",
              },
              Object {
                "prim": "FAILWITH",
              },
            ],
          ],
          "prim": "IF",
        },
        Object {
          "args": Array [
            Object {
              "prim": "operation",
            },
          ],
          "prim": "NIL",
        },
        Object {
          "prim": "PAIR",
        },
      ],
    ],
    "prim": "code",
  },
]
`;
